### ISSUE TITLE
Undo & Redo support

### ISSUE THREAD

**Issue by prasanthLalapeta** [daybrush/moveable#267]
## Environments
* Framework name: `react`
* Framework version: `16.13.1`
* Moveable Component version: `0.20.10`

## Description

Hi Daybrush

First of all kudos to your team. I just want to check with you. Do we have `undo` & `redo` feature support in library. If not any plans or suggestions that could be helpful for us.

Thanks, Prasanth

**Comment by daybrush** [comment]
@prasanthLalapeta 

It looks like an external function. Is there any reason to do that in moveable?

**Comment by prasanthLalapeta** [comment]
@daybrush 

Yes. If we are using moveable package in UI designing tool. Wherein actions like dragging, resizing, scaling and multi selection are available. 

It actually requires actions like `undo` , `redo` , `copy` and `paste` on elements.

It will be good if there is scope for above events in our package itself like we have `resize`, `drag` etc., because almost every product which performs `dragging`, `resizing` etc., obviously they need `undo`, `redo`, `copy` & `paste` features too :)

**Comment by happylinks** [comment]
In tella.tv we do this in our data structure (https://github.com/automerge/automerge), it natively has undo/redo and we implement copy and paste ourselves as well. Not sure how it would work in moveable, since it's only the dragging etc part, not the data layer? But for sure interesting :)

**Comment by daybrush** [comment]
@prasanthLalapeta  @happylinks 

https://daybrush.com/moveable/

in demo, the history function was implemented.
But this feature doesn't have to be in moveable.

https://github.com/daybrush/scena/blob/master/packages/react-editor/src/Editor/Viewport/MoveableMananger.tsx#L157-L200

**Comment by jadiagaurang** [comment]
@daybrush Can you provide the History library used to achieve Undo and Redo functions in the Demo at [https://daybrush.com/moveable/](https://daybrush.com/moveable/)? Also, can you help me identify which library was used to achieve Cut, Copy and Paste as well?

So, far I figured that you have used [keycon](https://github.com/daybrush/keycon) library to bind keyboard events with those library.

**Comment by daybrush** [comment]
@jadiagaurang 
There is no support plan.

A HistoryManager is created and implemented in the editor separately.

https://github.com/daybrush/scena/blob/master/packages/react-editor/src/Editor/managers/HistoryManager.ts
